File: go/leetcode/easy/Reverse Integer.go
// https://leetcode.com/problems/reverse-integer/
package main

import (
	"fmt"
	"math"
)

func abs(x int) int {
	if x > 0 {
		return x
	}

	return -x
}

func reverse(x int) int {
	num := 0

	for x != 0 {
		lastDigit := x % 10
		x = int(math.Trunc(float64(x / 10)))
		num = (num * 10) + lastDigit
	}

	if abs(num) > 2147483647 {
		return 0
	}

	return num
}

func main() {

	fmt.Println(math.Pow(2, 31))
}

File: js/leetcode/easy/Reverse Integer.js
// https://leetcode.com/problems/reverse-integer/
/**
 * @param {number} x
 * @return {number}
 */
const reverse = function (x) {
  let num = 0;
  while (x != 0) {
    const lastNum = x % 10;
    x = Math.trunc(x / 10);
    num = num * 10 + lastNum;
  }

  if (Math.abs(num) > Math.pow(2, 31)) {
    return 0;
  }

  return num;
};

// String manipulation
// const reverse = function (x) {
//   const num = +`${Math.abs(x)}`.split('').reverse().join('');

//   if (num > Math.pow(2, 31)) return 0;

//   return x < 0 ? -num : num;
// };
